frontend/public/components/configmap-and-secret-data.jsx

```diff
@@ -1,15 +1,83 @@
+import * as _ from 'lodash-es';
+
 import * as React from 'react';
+import * as PropTypes from 'prop-types';
+import { Heading } from './utils';
+
+export const ConfigMapData = ({data}) => {
+	const dl = [];
+	Object.keys(data || []).sort().forEach(k => {
+		dl.push(<dt key={`${k}-k`}>{k}</dt>);
+		dl.push(<dd key={`${k}-v`}><pre className="co-pre-wrap">{data[k]}</pre></dd>);
+	});
+
+	return <dl>{dl}</dl>;
+};
+
+export class SecretData extends React.PureComponent {
+	constructor(props) {
+		super(props);
+		this.state = { showSecret: false };
+		this.toggleSecret = this.toggleSecret.bind(this);
+	}
 
-const ConfigMapAndSecretData = ({data, decode}) => {
-  decode = decode || (v => v);
+	toggleSecret() {
+		this.setState({ showSecret: !this.state.showSecret })
+	}
 
-  const dl = [];
-  Object.keys(data || []).sort().forEach(k => {
-    dl.push(<dt key={`${k}-k`}>{k}</dt>);
-    dl.push(<dd key={`${k}-v`}><pre className="co-pre-wrap">{decode(data[k])}</pre></dd>);
-  });
+	render() {
```

Comment:
I think we will want to reduce `render()` down to focus on just `html` output as much as possible.  Some of this logic can be extracted out into some named functions.

Reply:
I wonder if you could restructure it to be more component oriented, something like this:

```JavaScript
  render() {
    const { data, type, decode } = this.props;
		const { showSecret } = this.state;
    return <React.Fragment>
			<Heading text="Data" >
				<a className="btn btn-link" onClick={this.toggleSecret}>{showSecret ? 'Hide Secret' : 'Reveal Secret'}</a>
			</Heading>
      <dl>
        {Object.keys(data).sort().forEach(key => {
          return isDockerConfig(key) ? <ComponentA {someProps}> : <ComponentB {otherProps}>     
        })}
      </dl>
		</React.Fragment>
  }
```

Reply:
Whoa, github messed that up 😦

Reply:
Makes sense ! Will try to decouple even more !